### PR TITLE
nomad: fix log file collection

### LINKED ISSUE #64: Nomad: incorrect pattern for log file copy
### Issue
Nomad log files are missing from the hcdiag bundle

### Reproduction steps
`hcdiag -nomad`

### Expected results
* `hcdiag` logging output shows log files as they are copied
* log files are present in the bundle

### Actual results
* log files are missing from the bundle
* `hcdiag` logging output contains copy command with incorrect pattern
```
2022-01-05T10:04:35.328-0500 [INFO]  hcdiag: running: seeker="copy /var/log/nomad//nomad.log-*"
```

### PR DESCRIPTION
**Pull request by davemay99**
* Fixes Nomad log copy pattern to correctly handle log file rollover
* Fixes issues with path concatenation on Windows

Resolves #64

### CODE REVIEW

**Comment by davemay99** [comment]
Fixed failing tests and bug in log path wildcard when a prefix is specified.

**Comment by davemay99** [comment]
Not sure why previous runs of golangci-lint didn't catch the redundant return, but this is now clean with all checks passing, and ready for review.

File: agent/agent.go [hunk 1]
```diff
@@ -263,15 +263,16 @@ func (a *Agent) WriteOutput() (err error) {
...
 	//  we don't waste users' time.
 	if mkdirErr := os.Mkdir(a.Config.Destination, 0755); mkdirErr != nil {
-		// TODO(mkcp): We will likely need more granular error handling here.
-		//  There are some cases where an error is a "happy path" - a dir exists, great, we can write to it.
-		//  But if the dir create fails and we proceed that's not ideal.
-		a.l.Debug("mkdir error", "error", err)
+		//  There are some cases where an error is a "happy path"
+		if errors.Is(mkdirErr, os.ErrExist) {
+			// a dir exists, great, we can write to it.
+			a.l.Trace("its just a harmless little bunny", "error", err)
```
Comment (gulducat) [inline agent/agent.go:269]: 🐇

File: client/nomad.go [hunk 2]
```diff
@@ -43,10 +44,11 @@ func GetNomadLogPath(api *APIClient) (string, error) {
 	// account for variable behavior depending on destination type
 	if _, file := filepath.Split(path); file == "" {
 		// this is a directory
-		path = path + "nomad-*"
+		path = path + "nomad*.log"
 	} else {
 		// this is a "prefix"
-		path = path + "-*"
+		path = strings.Replace(path, ".log", "", 1)
+		path = path + "*.log"
```
Comment (gulducat) [inline client/nomad.go:51]: this is interesting, I hadn't realized it varied in this way, good find!

File: util/copyfiles.go [hunk 3]
```diff
@@ -25,7 +26,12 @@ func CopyDir(to, src string) error {
 	absBase := filepath.Dir(absPath)
 
 	return filepath.Walk(src, func(path string, info os.FileInfo, err error) error {
-		target := filepath.Join(to, absBase, info.Name())
+		// Windows path may contain unsafe characters
+		targetMaybeUnsafe := filepath.Join(to, absBase, info.Name())
+
+		// TODO: more extensive path cleansing beyond handling C:\
+		target := strings.Replace(targetMaybeUnsafe, ":", "_", -1)
```
Comment (gulducat) [inline util/copyfiles.go:33]: wow hah, this is to account for flags like this yes?

```
hcdiag -includes C:\my-cool-log.log
```

**Review by gulducat — APPROVED** [review]
Great stuff, thanks for the contributions!

We'll get this into an upcoming release candidate soon!

heck yea windows

**Pull request merged**